### DISCUSSION TITLE
2.78 has "support for .NET 9", so should language features work?

### DISCUSSION THREAD

**Discussion by Marv51** [dotnet/docfx#10404] [General]
I updated to docfx 2.78.1 today. We also updated our projects to use the new C# 13 partial properties.
Docfx does not compile this successfully and requires a `<LangVersion>preview</LangVersion>` in the .csproj project.

Is this a bug, or is this the expected behavior? 

I can provide more details as needed, but this seems like a pretty general topic?

Example for partial property:
```csharp
[GeneratedRegex("abc|def")]
private static partial Regex AbcOrDefProperty { get; };
```
Example error message:
```
error CS8703: The modifier 'partial' is not valid for this item in C# 12.0. Please use language version 'preview' or greater.
```

**Comment by KalleOlaviNiemitalo** [comment]
C# 13 support would require upgrading the Microsoft.CodeAnalysis.* packages here: <https://github.com/dotnet/docfx/blob/618e71b424a265cff7cd140dfbafbe56a379371c/Directory.Packages.props#L29-L38>

Version 4.11.0 currently referenced there does not include <https://github.com/dotnet/roslyn/pull/74215>, which moved the C# 13 features from `LanguageVersion.Preview` to `LanguageVersion.CSharp13` in `src/Compilers/CSharp/Portable/Errors/MessageID.cs`.

Currently, the latest version of Microsoft.CodeAnalysis is [4.12.0-3.final](https://www.nuget.org/packages/Microsoft.CodeAnalysis/4.12.0-3.final), which is a prerelease version number; no 4.12.0 version has been released.  I guess that prevents dependabot from upgrading the packages automatically.

**Reply by Marv51** [reply]
Thanks! That actually makes a lot of sense.

**Reply by KalleOlaviNiemitalo** [reply]
These Roslyn packages were upgraded from 4.11.0 to 4.12.0 in <https://github.com/dotnet/docfx/pull/10448>.  That change came too late for the DocFX 2.78.2 release but it should be in the next one.
